lib/ansible/modules/windows/win_scheduled_task.ps1

```diff
@@ -165,7 +167,23 @@ try {
         Exit-Json $result
     }
 
-    $principal = New-ScheduledTaskPrincipal -UserId "$user" -LogonType ServiceAccount
+    # Handle RunAs/RunLevel options for the task
+
+    if (!$store_password) {
+        # Create a ScheduledTaskPrincipal for the task to run under
+        $principal = New-ScheduledTaskPrincipal -UserId $user -LogonType S4U -RunLevel $runlevel -Id Author
+        $registerRunOptionParams = @{Principal = $principal}
+    }
+    else {
+        # Specify direct credential and run-level values to add to Register-ScheduledTask
+        $registerRunOptionParams = @{
+            User = $user
+            RunLevel = $runlevel
+        }
+        if ($password) {
```

Comment:
So looking at the code this will occur when `store_password` is true. I would have thought to store the password you would need the password to be defined, is that not the case?